libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -790,6 +801,19 @@ public LoginOptions(String loginUrl, String passcodeHash, String oauthCallbackUr
             this.setJwt(jwt);
         }
 
+        public LoginOptions(String loginUrl, String passcodeHash, String oauthCallbackUrl,
+                            String oauthClientId, String[] oauthScopes, String clientSecret, String jwt,
+                            HashMap<String,String> additionalParameters) {
+            this(loginUrl, passcodeHash, oauthCallbackUrl, oauthClientId, oauthScopes,clientSecret,jwt);
+            this.setAdditionalParameters(additionalParameters);
+        }
+
+        public void setAdditionalParameters(HashMap<String,String> additionalParameters) {
+            this.additionalParameters = additionalParameters;
+            bundle.putSerializable(AuthenticatorService.KEY_ADDL_PARAMS,additionalParameters);
```

Comment:
@skyler-b Please take a look at Login options. I would have preferred to make all members of the class private, but the change made now is  a good compromise. I have deferred the creation of the bundle to on-demand. Directly manipulating the members or using setters yields the same result.